packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -1757,13 +1757,10 @@ class _CupertinoDatePickerMonthYearState extends State<CupertinoDatePicker> {
     double totalColumnWidths = 3 * _kDatePickerPadSize;
 
     for (int i = 0; i < columnWidths.length; i++) {
-      late final double offAxisFraction;
-      switch (i) {
-        case 0:
-          offAxisFraction = -0.3 * textDirectionFactor;
-        default:
-          offAxisFraction = 0.5 * textDirectionFactor;
-      }
+      final double offAxisFraction = switch (i) {
```

Comment:
The switch expression won't be able to provide exhaustiveness feedback for int values anyways. Consider using the ternary operator.

Reply:
done, thanks!

(I just did some other refactoring here too; hopefully it all looks good.)